src/main/resources/static/scripts/login.js

```diff
@@ -1,3 +1,31 @@
 function login() {
-    window.location.href = 'main.html';
+
+    var emailElement=document.getElementById("emailInput");
+    var passwordElement=document.getElementById("passInput");
+
+    var url = 'http://localhost:8080/auth/login';
+    var data = {"email": emailElement.value, "password": passwordElement.value};
+
+    fetch(url, {
+      method: 'POST',
+      body: JSON.stringify(data),
+      headers:{
+        'Content-Type': 'application/json'
+      }
+    }).then(res => {
+      if(res.ok) {
+        res.json().then(json => {
+          console.log('Success:', JSON.stringify(json));
+          window.location.href = 'main.html';
+        })
+      }
+      else {
+        res.json().then(json => {
+          console.log('Internal error:', JSON.stringify(json));
+          document.getElementById("error").innerHTML = "Invalid email or password!";
```

Comment:
If element is used more than once, assign it to variable.

Reply:
do you mean `json` element?

Reply:
I mean document.getElementById("error") element, for example 
`let errorElement = document.getElementById("error");`